blocksync/blocksync.go

```diff
@@ -184,10 +185,10 @@ func (bs *blockSyncer) ProcessSyncRequest(ctx context.Context, peer peerstore.Pe
 			return err
 		}
 		// TODO: send back multiple blocks in one shot
-		if err := bs.unicastHandler(context.Background(), peer,
-			blk.ConvertToBlockPb(),
-		); err != nil {
-			log.L().Debug("Failed to response to ProcessSyncRequest.", zap.Error(err))
+		syncCtx, cancel := context.WithTimeout(ctx, time.Second*5)
```

Comment:
Let's set it to 10s.
Add an early break on x times of error OR use the black list in p2p package (check details in the issue)